packages/tab/index.tsx

```diff
@@ -79,34 +114,41 @@ export default class Tab extends Component {
 
   get adapter() {
     return {
-      addClass: (className) => {
+      addClass: (className: string) => {
         const classList = new Set(this.state.classList);
         classList.add(className);
         this.setState({classList});
       },
-      removeClass: (className) => {
+      removeClass: (className: string) => {
         const classList = new Set(this.state.classList);
         classList.delete(className);
         this.setState({classList});
       },
-      hasClass: (className) => this.classes.split(' ').includes(className),
-      setAttr: (attr, value) => this.setState({[attr]: value}),
-      getOffsetLeft: () => this.tabElement_.current && this.tabElement_.current.offsetLeft,
-      getOffsetWidth: () => this.tabElement_.current && this.tabElement_.current.offsetWidth,
-      getContentOffsetLeft: () => this.tabContentElement_.current && this.tabContentElement_.current.offsetLeft,
-      getContentOffsetWidth: () => this.tabContentElement_.current && this.tabContentElement_.current.offsetWidth,
+      hasClass: (className: string) => this.classes.split(' ').includes(className),
+      setAttr: (attr: keyof TabState, value?: string | boolean) => (
```

Comment:
Is this the right typing?  Do you expect this to pass?

```typescript
.setAttr(previousIndicatorClientRect)
```

Does it?

Reply:
I mainly expect it to pass aria-selected or tabindex.